### DISCUSSION TITLE
stm32 lmic

### DISCUSSION THREAD

**Discussion by tcpipchip** [stm32duino/Arduino_Core_STM32#1584] [Q&A] [answered]
Sir,

I am using STM32L071 with lmic (LoRaWAN)...i can compile 100%

But, how can i use use Serial.print into the lmic routines to debug values ?

I am getting the following error

![image](https://user-images.githubusercontent.com/11488460/145558774-7ad395e6-0155-4af1-8277-1451e0eab17f.png)

**Comment by fpistm** [comment, marked as answer]
Well you can't as the Serial in a C++ instance object so you could not call it from C file.
You can try to use the core debug feature by defining `CORE_DEBUG` then include core_debug.h and use:
`void core_debug(const char *format, ...)`

**Comment by tcpipchip** [comment]
Thank you so much @fpistm

**Comment by tcpipchip** [comment]
it´s works!
